gaia-toolbar.js

```diff
@@ -15,291 +14,361 @@ var GaiaDialogMenu = require('gaia-dialog-menu');
+    this.style.visibility = '';
+  },
+
+  getOverflowData: function() {
+    var space = this.els.inner.clientWidth;
+    var child = this.children[0];
+    var overflowed = false;
+    var willFit = 0;
+    var total = 0;
+    var overflow;
+    var width;
+
+    while (child) {
+      width = child.clientWidth;
+      overflow = width + total - space;
+
+      if (overflow > 3) {
+        overflowed = true;
+        break;
+      }
+
+      willFit++;
+      total += width;
+      child = child.nextElementSibling;
+    }
 
-  border-top: 1px solid;
-  border-color:
-    var(--border-color,
-    var(--background-plus));
+    return {
+      willFit: willFit,
+      remaining: space - total,
+      overflowed: overflowed
+    };
+  },
+
+  onOverflowed: function(overflow) {
+    var items = slice.call(this.children, 0, -1);
+    var minMoreButtonWidth = 70;
+    var extra = overflow.remaining < minMoreButtonWidth ? 1 : 0;
+    var numToHide = items.length - overflow.willFit + extra;
+
+    this.hiddenChildren = slice.call(items, 0 - numToHide);
+    this.hiddenChildren.forEach(el => el.classList.add('overflowing'));
+    this.els.inner.classList.add('overflowed');
+  },
+
+  release: function() {
+    if (!this.hiddenChildren) { return; }
+    this.els.inner.classList.remove('overflowed');
+    forEach.call(this.hiddenChildren, function(el) {
+      el.classList.remove('overflowing');
+    });
+  },
+
+  getTotalItemLength: function() {
+    return [].reduce.call(this.children, function(total, el) {
+      return total + el.clientWidth;
+    }, 0);
+  },
+
+  shadowStyleHack: function() {
+    var style = this.shadowRoot.querySelector('style').cloneNode(true);
+    this.classList.add('-content', '-host');
+    style.setAttribute('scoped', '');
+    this.appendChild(style);
+  },
+
+  openOverflow: function(e) {
+    this.dialog = new GaiaDialogMenu();
+
+    this.hiddenChildren.forEach(function(el) {
+      this.dialog.appendChild(el);
+      el.classList.remove('overflowing');
+    }, this);
+
+    this.els.moreButton.setAttribute('aria-expanded', true);
+
+    this.appendChild(this.dialog);
+    this.dialog.addEventListener('click', this.dialog.close.bind(this.dialog));
+    this.dialog.addEventListener('closed', this.onDialogClosed.bind(this));
+    this.dialog.open(e);
+  },
+
+  onDialogClosed: function() {
+    this.dialog.remove();
+    this.dialog = null;
+
+    this.els.moreButton.setAttribute('aria-expanded', false);
+
+    this.hiddenChildren.forEach(function(el) {
+      el.classList.add('overflowing');
+      this.insertBefore(el, this.lastChild);
+    }, this);
+  },
+
+  attrs: {
+    l10nMoreId: {
+      get: function() { return this._l10nMoreId; },
+      set: function(value) {
+        if (this._l10nMoreId === value) { return; }
+        this._l10nMoreId = value;
+        if (value) {
+          this.setAttribute('l10n-more-id', value);
+          this.els.moreButton.setAttribute('data-l10n-id', value);
```

Comment:
AFAIK l10n.js will not localize `data-l10n0id` nodes inside the shadow-root without using `.localizeFragment()`, no?